crates/trie/parallel/src/proof_task.rs

```diff
@@ -158,57 +434,135 @@ where
         self.executor.spawn_blocking(move || match task {
-            ProofTaskKind::StorageProof(input, sender) => {
-                proof_task_tx.storage_proof(input, sender, tx_sender);
-            }
             ProofTaskKind::BlindedAccountNode(path, sender) => {
                 proof_task_tx.blinded_account_node(path, sender, tx_sender);
             }
-            ProofTaskKind::BlindedStorageNode(account, path, sender) => {
-                proof_task_tx.blinded_storage_node(account, path, sender, tx_sender);
+            // Storage trie operations should never reach here as they're routed to worker pool
+            ProofTaskKind::BlindedStorageNode(_, _, _) | ProofTaskKind::StorageProof(_, _) => {
+                unreachable!("Storage trie operations should be routed to worker pool")
             }
         });
 
         Ok(())
     }
 
     /// Loops, managing the proof tasks, and sending new tasks to the executor.
+    ///
+    /// # Task Routing
+    ///
+    /// - **Storage Trie Operations** (`StorageProof` and `BlindedStorageNode`): Routed to
+    ///   pre-spawned worker pool via unbounded channel. Only falls back to `pending_tasks` if
+    ///   workers are disconnected (e.g., all workers panicked).
+    /// - **Account Trie Operations** (`BlindedAccountNode`): Queued for on-demand execution via
+    ///   `pending_tasks`.
+    ///
+    /// # Shutdown
+    ///
+    /// On termination, `storage_work_tx` is dropped, closing the channel and
+    /// signaling all workers to shut down gracefully.
     pub fn run(mut self) -> ProviderResult<()> {
         loop {
             match self.proof_task_rx.recv() {
-                Ok(message) => match message {
-                    ProofTaskMessage::QueueTask(task) => {
-                        // Track metrics for blinded node requests
-                        #[cfg(feature = "metrics")]
-                        match &task {
-                            ProofTaskKind::BlindedAccountNode(_, _) => {
-                                self.metrics.account_nodes += 1;
+                Ok(message) => {
+                    match message {
+                        ProofTaskMessage::QueueTask(task) => match task {
+                            ProofTaskKind::StorageProof(input, sender) => {
+                                match self.storage_work_tx.send(StorageWorkerJob::StorageProof {
+                                    input,
+                                    result_sender: sender,
+                                }) {
+                                    Ok(_) => {
+                                        tracing::trace!(
+                                            target: "trie::proof_task",
+                                            "Storage proof dispatched to worker pool"
+                                        );
+                                    }
+                                    Err(crossbeam_channel::SendError(job)) => {
```

Comment:
Another case of "can this actually happen?" We know that the worker pool was successfully initialized, and each worker cannot break out of its loop without Terminate having been called, and the channel is unbounded. If it's not really possible for this to happen then I say let's `self.storage_work_tx.send(...).expect(...)`

Reply:
makes sense, let me address this in the next PR (i have already made some breaking change there based on this)! thanks for the note.

Reply:
makes sense, let me address this in the next PR (i have already made some breaking change there based on this)! thanks for the note.